### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by killme**
Turning this into a PR as it is a somewhat large diff and turns on warnings in maven (which will let builds with warnings count as an unsuccessful build CI wise).

The main thing changed in this PR is the following:
- Enable logging of warnings
- Fix the warnings that popped up and fixed minor bugs here and there. 
- Move some of the "deprecated" bukkit calls to a separate class that deals with it in one place instead of having `@SuppressWarnings("deprecated")` all over the code-base. (Existing code is not adjusted to use this apart from the call-sites with warnings)

If this gets merged in please all make sure to enable warnings in your compiler.

**TL;DR** Fixed warnings; please make sure I didn't do anything stupid with my "fixes"

### CODE REVIEW

File: src/main/java/com/elmakers/mine/bukkit/action/builtin/AnimateAction.java [hunk 1]
```diff
@@ -22,774 +20,4 @@ public SpellResult step(CastContext context) {
 	};
 
 	public static boolean DEBUG = false;
-
```
Comment (NathanWolf) [inline src/main/java/com/elmakers/mine/bukkit/action/builtin/AnimateAction.java]: Why the big chop here?

This was admittedly a big TODO that never got done, just curious why it had to be stripped.

File: src/main/java/com/elmakers/mine/bukkit/action/builtin/SimulateAction.java [hunk 2]
```diff
@@ -3,789 +3,17 @@
...
 import com.elmakers.mine.bukkit.api.spell.SpellResult;
-import com.elmakers.mine.bukkit.block.MaterialAndData;
-import org.bukkit.Location;
-import org.bukkit.Material;
-import org.bukkit.block.Block;
-import org.bukkit.block.BlockFace;
-import java.util.ArrayList;
-import java.util.List;
 
 public class SimulateAction extends BaseSpellAction {
```
Comment (NathanWolf) [inline src/main/java/com/elmakers/mine/bukkit/action/builtin/SimulateAction.java]: Same question/comment as AnimateAction - these were meant to replace AnimateSpell and SimulateSpell, but I never went anywhere with them.

**Comment by NathanWolf** [comment]
I looked over all of this about as best I could right now, at this point I trust you!

Thanks much for all your cleanup and maintainance.

File: src/main/java/com/elmakers/mine/bukkit/action/builtin/AnimateAction.java (hunk 1 above)
Comment (killme) [inline src/main/java/com/elmakers/mine/bukkit/action/builtin/AnimateAction.java]: Dead / commented out code isn't exactly useful either when we have git though.
I just removed the dead lines because of unused warnings and then the commented out code out of cleaning habit.
